### ISSUE TITLE
Introduce a way to opt-out of source info in Diagnostics StackFrame 

### ISSUE THREAD

**Issue by eerhardt** [dotnet/runtime#34910]
The source line info support for System.Diagnostics.StackTrace/StackFrame comes with a noticeable cost for workloads which are size sensitive and don't contain PDB information anyway.

With change https://github.com/dotnet/runtime/pull/32023, we added a `PreserveDependency` attribute here:

https://github.com/dotnet/runtime/blob/768c6fc99c099465491fbf0185b145581243f130/src/coreclr/src/System.Private.CoreLib/src/System/Diagnostics/StackFrameHelper.cs#L87-L93

As noted in the PR, the change added ~200K to the "Console Hello World" app, bringing in the following assemblies, which were not needed before:

![image](https://user-images.githubusercontent.com/8291187/79147084-053b8600-7d89-11ea-94ec-6d2c6c797aab.png)

We should consider adding a feature switch (dotnet/designs#99) or similar functionality to allow app developers to opt-out of this behavior.

cc @vitek-karas @jkotas @stephentoub @marek-safar

**Comment by danmosemsft** [comment]
Added to Linker project which I guess is how we keep track of all such work.

**Comment by marek-safar** [comment]
I don't think this should be feature switch in the design sense as above but illinker property which controls debugging support.

**Comment by krwq** [comment]
@eerhardt I'm marking this as 5.0 so it doesn't disappear from the radar, please change milestone to Future if you feel like this is not important for 5.0

**Comment by eerhardt** [comment]
@marek-safar @vargaz - I just discovered that this code isn't used on Mono's CoreLib. Instead, I see that Mono has a native C reader for PPDBs.

https://github.com/dotnet/runtime/blob/b24e9cd0996c688fd627fc1698986097e54103cd/src/mono/netcore/System.Private.CoreLib/src/System/Diagnostics/StackFrame.Mono.cs#L51-L53

Is this the long-term solution for Mono? I'm wondering how important it is for us to solve the above Diagnostics StackFrame issue in .NET 5. If Blazor isn't going to be affected by it, it might not be critical to solve.

**Comment by marek-safar** [comment]
I think we'll investigate using managed ppds reader but we might not be able to do it for .NET5 but long term this looks attractive to me.

#32023 is not relevant to size sensitive workloads and will become only once the logic is moved to shared SPC. What is however still relevant to us is the "forced" [StackTraceSymbol](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.StackTrace/src/ILLinkTrim.xml#L4) persistence in S.D.S which I'm not sure why it's still there

**Comment by eerhardt** [comment]
> What is however still relevant to us is the "forced" StackTraceSymbol persistence in S.D.S which I'm not sure why it's still there

I'll look into removing it today. Thanks for the pointer.

**Comment by eerhardt** [comment]
This is not necessary for Blazor WASM in 5.0. Moving to 6.0.

**Comment by jkotas** [comment]
This should be tagged on `StackTraceSupport` introduced in https://github.com/dotnet/runtime/pull/88235 .

**Comment by pavelsavara** [comment]
Is this implemented by `System.Diagnostics.StackTrace.IsSupported` and `System.Diagnostics.StackTrace.IsLineNumberSupported` feature flags ? Or is something missing ?

https://github.com/dotnet/runtime/pull/123429

**Comment by jkotas** [comment]
> Is this implemented by System.Diagnostics.StackTrace.IsSupported and System.Diagnostics.StackTrace.IsLineNumberSupported feature flags ? 

I think so
